packages/neon/neon/lib/src/widgets/list_view.dart

```diff
@@ -1,66 +1,79 @@
+    this.topFixedChildren,
+    this.topScrollingChildren,
+    super.key,
+  }) : sliver = SliverList.builder(
+          itemCount: itemCount,
+          itemBuilder: itemBuilder,
+        );
+
+  const NeonListView.custom({
+    required this.isLoading,
+    required this.error,
+    required this.onRefresh,
+    required this.sliver,
     this.scrollKey,
-    this.withFloatingActionButton = false,
     this.topFixedChildren,
     this.topScrollingChildren,
     super.key,
   });
 
-  final List<T>? items;
   final bool isLoading;
-  final dynamic error;
-  final Future Function() onRefresh;
-  final Widget Function(BuildContext, T data) builder;
+  final Object? error;
+  final RefreshCallback onRefresh;
   final String? scrollKey;
-  final bool withFloatingActionButton;
   final List<Widget>? topFixedChildren;
   final List<Widget>? topScrollingChildren;
+  final Widget sliver;
 
   @override
-  Widget build(final BuildContext context) => RefreshIndicator(
-        onRefresh: onRefresh,
-        child: Column(
-          children: [
-            ...?topFixedChildren,
-            NeonLinearProgressIndicator(
-              margin: const EdgeInsets.symmetric(
-                horizontal: 10,
-                vertical: 5,
+  Widget build(final BuildContext context) {
+    final refreshIndicatorKey = GlobalKey<RefreshIndicatorState>();
+    final hasFloatingActionButton = Scaffold.maybeOf(context)?.hasFloatingActionButton ?? false;
+
+    return RefreshIndicator.adaptive(
+      key: refreshIndicatorKey,
+      onRefresh: onRefresh,
+      child: Scrollbar(
+        child: CustomScrollView(
+          key: scrollKey != null ? PageStorageKey<String>(scrollKey!) : null,
+          primary: true,
+          slivers: [
+            if (topFixedChildren != null)
+              SliverList.builder(
+                itemCount: topFixedChildren!.length,
+                itemBuilder: (final context, final index) => topFixedChildren![index],
               ),
-              visible: isLoading,
-            ),
-            Expanded(
-              child: Scrollbar(
-                child: ListView(
-                  primary: true,
-                  key: scrollKey != null ? PageStorageKey<String>(scrollKey!) : null,
-                  padding: withFloatingActionButton ? const EdgeInsets.only(bottom: 88) : null,
-                  children: [
-                    ...?topScrollingChildren,
-                    NeonException(
-                      error,
-                      onRetry: onRefresh,
-                    ),
-                    if (items != null) ...[
-                      for (final item in items!) ...[
-                        builder(context, item),
-                      ],
-                    ],
-                  ],
+            if (error != null)
+              SliverToBoxAdapter(
+                child: NeonException(
+                  error,
+                  onRetry: () async => refreshIndicatorKey.currentState!.show(),
```

Comment:
This needs to actually trigger the refresh, right? I would also prefer if there just is a listener for the result stream and depending on isLoading it would show or hide the refresh indicator.

Edit: I just realized this is probably what you meant when you said you still need to implement isLoading

Reply:
The documentstion arround the `RefreshInicatorState.show()` method is a bit confusing.

Afaict it will trigger the normal pull down. Showing the loading animation and calling `onRefresh`.

As you mentioned I'd like to wire the initial `isLoading` into this causing less rebuilds and a better UI.